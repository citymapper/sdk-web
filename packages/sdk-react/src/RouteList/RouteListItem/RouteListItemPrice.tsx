import type { Price } from '@citymapper/api/@types'
import React from 'react'
import styled from '../../utils/cmStyled'

export interface RouteListItemPriceProps {
  price: Price
}

const Wrapper = styled('div', { className: 'cm-route-price-root' })`
  font-weight: 600;
  font-size: 0.8125em;
  line-height: 0.983125em;
  margin-top: 2px;
  margin-left: ${({ theme }) => theme.spacing(1)};
  margin-right: ${({ theme }) => theme.spacing(1)};
  color: ${({ theme }) => theme.palette.brand.primary.color};

  &.cm-route-price {
    ${({ theme }) => theme.CmRoutePrice?.root}
  }
`

const RouteListItemPrice: React.FC<RouteListItemPriceProps> = ({ price }) => {
  return <Wrapper>{price.formatted}</Wrapper>
}

export default RouteListItemPrice
