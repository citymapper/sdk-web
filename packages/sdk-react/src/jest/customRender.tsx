import { render, RenderOptions } from '@testing-library/react'
import React, { JSXElementConstructor, ReactElement } from 'react'
import CMProvider from '../CMProvider'

const TestWrapper: React.FC<{ children: React.ReactElement }> = ({
  children,
}) => (
  <CMProvider
    config={{
      apiKey: '',
      citymapperFont: false,
    }}
  >
    {children}
  </CMProvider>
)

const customRender = (
  ui: ReactElement<any, string | JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import('@testing-library/dom/types/queries'),
    HTMLElement,
    HTMLElement
  >
) => render(ui, { wrapper: TestWrapper, ...options })

export default customRender
