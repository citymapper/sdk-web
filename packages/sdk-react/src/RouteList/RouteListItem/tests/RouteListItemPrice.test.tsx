import React from 'react'
import customRender from '../../../jest/customRender'
import RouteListItemPrice from '../RouteListItemPrice'

describe('RouteListItemPrice', () => {
  it('should match snapshot', () => {
    const { container } = customRender(
      <RouteListItemPrice
        price={{ amount: '2.50', currency: 'GBP', formatted: 'Â£2.50' }}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
