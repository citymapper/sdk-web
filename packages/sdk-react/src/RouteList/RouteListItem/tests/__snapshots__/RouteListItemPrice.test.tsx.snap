// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteListItemPrice should match snapshot 1`] = `
.emotion-0 {
  font-weight: 600;
  font-size: 0.8125em;
  line-height: 0.983125em;
  margin-top: 2px;
  margin-left: 8px;
  margin-right: 8px;
  color: #0f69f0;
}

<div>
  <div
    class="cm-route-price-root emotion-0"
  >
    Â£2.50
  </div>
</div>
`;
